### PR TITLE
Add Starkli-Scarb interaction using katana

### PR DESCRIPTION
**Pull request by gianalarcon**
Based on the PR #175
Closes issue #67

- [x] Scarb
- [x] Starkli

@ponderingdemocritus @glihm I will need your feedback here

### CODE REVIEW

File: docs/pages/toolchain/katana/interact.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+```console
+scarb new vote
+```
+
+Add contract dependencies to scarb.toml
+
+```toml
+[dependencies]
+starknet = "2.5.0"
+[[target.starknet-contract]]
```
Comment (glihm) [inline docs/pages/toolchain/katana/interact.md:67]:
```suggestion

[[target.starknet-contract]]
```

File: docs/pages/toolchain/katana/interact.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+```
+
+Copy the vote contract to lib.cairo
```
Comment (glihm) [inline docs/pages/toolchain/katana/interact.md:70]: This contract is interesting, but is huge IMHO.
What do you think about having a simple counter or something like this? Or perhaps you've some other cases you want to add?

File: docs/pages/toolchain/katana/interact.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+2. Compile contract
+
+```bash
+scarb build
+```
+
+3. Environment variables
+
+   Having compiled the smart contract, it's time to declare it with Starkli and katana. For convenient management, place the following environment variables in a .env file within the `src/` directory.
```
Comment (glihm) [inline docs/pages/toolchain/katana/interact.md:258]: We've already defined those variables at the beginning of the tutorial. Seems a better approach to actually use the `.env`. We may refer to the starkli book for the first part then, and keep this one as it's more intuitive and re-usable?

File: docs/pages/toolchain/katana/interact.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+Cairo traceback (most recent call last):
+Unknown location (pc=0:67)
+Unknown location (pc=0:1835)
+Unknown location (pc=0:2478)
+Unknown location (pc=0:3255)
+Unknown location (pc=0:3795)
+
+Error in the called contract (0x02c44f2d396fc5f9caa551e8c1d901d943a3b8cc5c433c88a1bf10b1f15fcd15):
+Execution failed. Failure reason: 0x555345525f414c52454144595f564f544544 ('USER_ALREADY_VOTED').
+```
```
Comment (glihm) [inline docs/pages/toolchain/katana/interact.md:368]: That's nice to show a failure. :+1:

**Review by glihm — COMMENTED** [review]
@gianalarcon good job on that, a first review here. :+1:

File: docs/pages/toolchain/katana/interact.md (hunk 2 above)
Comment (gianalarcon) [reply docs/pages/toolchain/katana/interact.md:71]: You are right. I suggest we can show a simple/basic example with a `counter contract`. Then we also can show an intermediate example with this `voting contract`

File: docs/pages/toolchain/katana/interact.md (hunk 3 above)
Comment (gianalarcon) [reply docs/pages/toolchain/katana/interact.md:259]: Yes. I was thinking how to explain well and where to keep the environment variables. I am gonna follow your suggestion on this.

File: docs/pages/toolchain/katana/interact.md (hunk 2 above)
Comment (glihm) [reply docs/pages/toolchain/katana/interact.md:71]: Yep sounds good! The idea is as some reader may just dive into starknet, having something simpler first would not discourage them. :)

File: docs/pages/toolchain/katana/interact.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://docs.swmansion.com/scarb/install.sh | sh
+```
+
+You can check your installation by running `scarb --version`, then you will get the starkli version.
+
+To install `Katana`, use the `dojoup` installer from the command line:
+
+```bash
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:26]: could you move these to a 'snippets' directory and we can reuse the snippet everywhere. https://vocs.dev/docs/guides/code-snippets

Also could you do it for other CLI commands that are generalised

File: docs/pages/toolchain/katana/interact.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+## Basics of Katana and Starkli
+
+### katana sequencer
+
+```console
+katana --disable-fee
+```
+
+After starting the node, a list of accounts will be automatically generated and deployed.
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:41]:
```suggestion
After starting the Katana, a list of accounts will be automatically generated and deployed.
```

File: docs/pages/toolchain/katana/interact.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+### Starkli built-in accounts and configuration
+
+Starkli supports a list of built-in accounts for katana. These built-in accounts are for local development and no one should be using them for anything serious anyways. For example the address of the `Katana-0` built-in account is `0x6162896d1d7ab204c7ccac6dd5f8e9e7c25ecd5ae4fcb4ad32e57786bb46e03`. You can check the full list of account addresses [here](https://github.com/xJonathanLEI/starkli/blob/e9a28f1b6e37bcc9fc53b7b7130e935894856739/src/account.rs#L76).
+
+We will need this `katana-0` built-in account later when we define our `STARKNET_ACCOUNT`. For more details about accounts, please refer to the accounts [page](https://book.starkli.rs/accounts).
+
+## Contract Deployment and Interaction
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:49]: add context here that this is a new directory.

File: docs/pages/toolchain/katana/interact.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+After running, expect an output similar to:
+
+```console
+[
+    "0x0000000000000000000000000000000000000000000000000000000000000000"
+]
+```
+
+It means the vale of `stored_data` is cero.
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:177]:
```suggestion
It means the vale of `stored_data` is zero.
```

File: docs/pages/toolchain/katana/interact.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+            self.stored_data.write(x);
+        }
+        fn get(self: @ContractState) -> u128 {
+            self.stored_data.read()
+        }
+    }
+}
+```
+
+### Compile contract and Environment variables setup
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:97]:
```suggestion
### Compile contract and environment variables setup
```

Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:97]:
```suggestion
### Compile contract and add environment variables
```

File: docs/pages/toolchain/katana/interact.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+Compile your contract using scarb
+
+```bash
+scarb build
+```
+
+Having compiled the smart contract, it's time to declare it with Starkli and katana. For convenient management, place the following environment variables in a .env file within the `src/` directory.
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:105]:
```suggestion
Having compiled the smart contract, it's time to declare it with Starkli and katana. For clean enviroment management, place the following environment variables in a .env file within the `src/` directory.
```

File: docs/pages/toolchain/katana/interact.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+To declare your contract, execute:
+
+```bash
+starkli declare target/dev/simple_storage_SimpleStorage.contract_class.json
+```
+
+Upon successful command execution, you'll obtain a contract class hash: This unique hash serves as the identifier for your contract class within Starknet. For example:
+
+```sh
+Class hash declared: 0x07ad2516dd66fb2e274e78d4357837cad689c9fffaa347feb9800b231b37b306
```
Comment (ponderingdemocritus) [inline docs/pages/toolchain/katana/interact.md:137]: do the full printout here and focus on the class hash

**Pull request merged**